### PR TITLE
Solves #2003. Add specific code to delete volumes created by statefulsets when an okteto destroy is executed

### LINKED ISSUE #2003: Okteto destroy does not delete pvc created by a statefulset
**Describe the bug**
okteto destroy command does not destroy volumes mounted to a statefulset and created directly by k8s when the statefulset is created

**To Reproduce**

1. Deploy this stack https://github.com/okteto/flask-producer-consumer
2. Delete everything
3. A couple of PVCs are left behind

**Expected behavior**
All PVCs created should be removed if the flag to include volumes is selected

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by ifbyol**
Signed-off-by: Nacho Fuertes <nacho@okteto.com>

Fixes #2003

## Proposed changes

- Include specific code to detect all PVCs created for statefulsets and delete them before delete anything else

### CODE REVIEW

File: cmd/deploy/deploy.go [hunk 1]
```diff
@@ -289,6 +289,7 @@ func getProxyHandler(name, token string, clusterConfig *rest.Config) (http.Handl
 				return
 			}
 
+			defer r.Body.Close()
```
Comment (ifbyol) [inline cmd/deploy/deploy.go:292]: This was missing and it could generate some performance issue in the proxy

File: cmd/destroy/destroy.go [hunk 2]
```diff
@@ -160,6 +161,14 @@ func (dc *destroyCommand) runDestroy(ctx context.Context, cwd string, opts *Opti
 		return err
 	}
 	deployedBySelector := labels.NewSelector().Add(*deployedByLs).String()
+	deleteOpts := namespaces.DeleteAllOptions{
+		LabelSelector:  deployedBySelector,
+		IncludeVolumes: opts.DestroyVolumes,
+	}
+
+	if err := dc.nsDestroyer.DestroySFSVolumes(ctx, opts.Namespace, deleteOpts); err != nil {
```
Comment (ifbyol) [inline cmd/destroy/destroy.go:169]: This has to be done before we delete things with helm

File: pkg/k8s/volumes/crud.go [hunk 3]
```diff
@@ -192,6 +192,21 @@ func Destroy(ctx context.Context, name, namespace string, c *kubernetes.Clientse
 
 }
 
+// DestroyWithoutTimeout destroys a PVC without checking if it was detached or not
+func DestroyWithoutTimeout(ctx context.Context, name, namespace string, c kubernetes.Interface) error {
```
Comment (ifbyol) [inline pkg/k8s/volumes/crud.go:196]: In this case we don't have to wait until the volume is detached, so why it was called `WithoutTimeout`

**Review by ifbyol — COMMENTED** [review]
When we destroy things, it could generate some annoying warnings regarding some deprecated APIs,

```
W1123 11:29:58.369746   91041 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W1123 11:29:58.369922   91041 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W1123 11:29:58.415528   91041 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W1123 11:29:58.456929   91041 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W1123 11:29:58.538229   91041 warnings.go:70] rbac.authorization.k8s.io/v1beta1 Role is deprecated in v1.17+, unavailable in v1.22+; use rbac.authorization.k8s.io/v1 Role
W1123 11:29:58.589557   91041 warnings.go:70] rbac.authorization.k8s.io/v1beta1 RoleBinding is deprecated in v1.17+, unavailable in v1.22+; use rbac.authorization.k8s.io/v1 RoleBinding
```

It seems it is logged by `client-go` but I was not able to mute it as I did with logrus. It seems client-go uses klog and I didn't find a way to change the level and restore it after.

Have we done this before with that log?

**Review by pchico83 — APPROVED** [review]

**Pull request merged**